src/System.Text.Primitives/System/Text/EncodingData.cs

```diff
@@ -365,8 +345,153 @@ private bool VerifySuffix(ReadOnlySpan<byte> buffer, int codeUnitIndex, Symbol s
         // CC = the max value, and DD = the max value's index in the same coord-system as BB.
         struct ParsingTrieNode
         {
-            public byte valueOrNumChildren;
+            public byte ValueOrNumChildren;
             public int IndexOrSymbol;
         }
+        
+        private static int CompareSuffix(Suffix x, Suffix y)
+        {
+            int index = 0;
+            if (x.Bytes.Length == 0 || y.Bytes.Length == 0)
+            {
+                throw new ArgumentException("Symbol cannot be zero bytes long");
+            }
+            while (true)
+            {
+                if (index == x.Bytes.Length)
+                {
+                    if (index == y.Bytes.Length)
+                    {
+                        throw new ArgumentException(String.Format("Symbols cannot be identical"));
+                    }
+                    throw new ArgumentException("Symbols are ambiguous");
+                }
+                if (index == y.Bytes.Length)
+                {
+                    throw new ArgumentException("Symbols are ambiguous");
+                }
+                int compareResult = x.Bytes[index].CompareTo(y.Bytes[index]);
+                if (compareResult != 0)
+                {
+                    return compareResult;
+                }
+                index++;
+            }
+        }
+
+        private struct Suffix
+        {
+            public int SymbolIndex;
+            public ReadOnlySpan<byte> Bytes;
+
+            public Suffix(int symbolIndex, ReadOnlySpan<byte> bytes)
+            {
+                SymbolIndex = symbolIndex;
+                Bytes = bytes;
+            }
+        }
+
+        private struct SuffixClump
+        {
+            public byte BeginningByte;
+            public List<Suffix> Suffixes;
+
+            public SuffixClump(byte beginningByte)
+            {
+                BeginningByte = beginningByte;
+                Suffixes = new List<Suffix>();
```

Comment:
The capacity required would be between 1 and the number of symbols, which is currently 17. A capacity of 20 or so would guarantee that no resizing is needed. I can make the ctor set this, like: Suffixes = new List<Suffix>(20);